lib/util/util.CombiningMessageProvider.js

```diff
@@ -0,0 +1,53 @@
+this.util = this.util || {};
+
+util.CombiningMessageProvider = ( function() {
+	'use strict';
+
+	/**
+	 * Providing messages using two message providers
+	 * @class util.CombiningMessageProvider
+	 * @licence GNU GPL v2+
+	 * @author H. Snater < mediawiki@snater.com >
+	 * @author Adrian Heine < adrian.heine@wikimedia.de >
+	 *
+	 * @constructor
+	 *
+	 * @param {util.MessageProvider} a
+	 * @param {util.MessageProvider} b
+	 */
+	function CombiningMessageProvider( a, b ) {
```

Comment:
1. I find the name a bit confusing. To me it sounds like it does some concatenation, but it's an alternative.
2. Why is this limited to 2 and not an array?